dm/ctl/master/operate_task.go

```diff
@@ -0,0 +1,184 @@
+
+func addOperateSourceTaskFlags(cmd *cobra.Command) {
+	// control workload to dm-cluster for sources with large number of tasks.
+	cmd.Flags().Int(batchSizeFlag, defaultBatchSize, "batch size")
+}
+
+func operateSourceTaskFunc(taskOp pb.TaskOp, cmd *cobra.Command) error {
+	source, batchSize, ok := parseOperateSourceTaskParams(cmd)
+	if !ok {
+		cmd.SetOut(os.Stdout)
+		common.PrintCmdUsage(cmd)
+		return errors.New("please check output to see error")
+	}
+
+	sources := []string{source}
+	ctx, cancel := context.WithTimeout(context.Background(), common.GlobalConfig().RPCTimeout)
+	defer cancel()
+
+	req := pb.QueryStatusListRequest{Sources: sources}
+	resp := &pb.QueryStatusListResponse{}
+	if err := common.SendRequest(ctx, "QueryStatus", &req, &resp); err != nil {
+		common.PrintLinesf("cannot query status of source: %v", sources)
+		return err
+	}
+
+	if !resp.Result || len(resp.Sources) == 0 {
+		common.PrettyPrintInterface(&batchTaskResult{Result: false, Msg: resp.Msg, BatchResults: []*operateTaskResult{}})
+		return nil
+	}
+
+	result := batchOperateTask(taskOp, batchSize, sources, resp.Sources[0].SubTaskStatus)
+	common.PrettyPrintInterface(result)
+
+	return nil
+}
+
+func batchOperateTask(taskOp pb.TaskOp, batchSize int, sources []string, subTaskStatus []*pb.SubTaskStatus) *batchTaskResult {
+	result := batchTaskResult{Result: true, BatchResults: []*operateTaskResult{}}
+
+	if len(subTaskStatus) < batchSize {
+		batchSize = len(subTaskStatus)
+	}
+
+	workCh := make(chan string)
+	go func() {
+		for _, subTask := range subTaskStatus {
+			workCh <- subTask.Name
+		}
+		close(workCh)
+	}()
+
+	var wg sync.WaitGroup
+	resultCh := make(chan *operateTaskResult, 1)
+	for i := 0; i < batchSize; i++ {
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+
+			for name := range workCh {
+				taskResult := operateTaskResult{TaskName: name, Op: taskOp.String()}
+				taskOpResp, err := common.OperateTask(taskOp, name, sources)
+				if err != nil {
+					taskResult.Result = false
+					taskResult.Msg = err.Error()
+				} else {
+					taskResult.Result = true
+					taskResult.Msg = taskOpResp.Msg
+					taskResult.Sources = taskOpResp.Sources
+				}
+				resultCh <- &taskResult
+			}
+		}()
+	}
+
+	go func() {
+		wg.Wait()
+		close(resultCh)
+	}()
+
+	for item := range resultCh {
+		result.BatchResults = append(result.BatchResults, item)
+	}
+
+	sort.Slice(result.BatchResults, func(i, j int) bool {
+		return result.BatchResults[i].TaskName < result.BatchResults[j].TaskName
+	})
+
+	return &result
+}
+
+func parseOperateSourceTaskParams(cmd *cobra.Command) (string, int, bool) {
+	sources, err := common.GetSourceArgs(cmd)
+	if err != nil {
+		return "", 0, false
+	}
+	if len(sources) == 0 {
+		common.PrintLinesf(`must give one source-name when task-name/task-conf is not specified`)
+		return "", 0, false
+	} else if len(sources) > 1 {
+		common.PrintLinesf(`can give only one source-name when task-name/task-conf is not specified`)
+		return "", 0, false
+	}
```

Comment:
Is is possible for us to support `len(sources) > 1` in the future?

Reply:
maybe when we have some feedback from users, it's quite enough for now.

PS: with multiple `-s` parameter, the semantic is a little confusing, it can be:
- tasks with all sources
- tasks with either source